### ISSUE TITLE
Keras model for sentiment analysis is not learning anything!

### ISSUE THREAD

**Issue by tirthasheshpatel** [tensorflow/tensorflow#37872]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information** 

- Google Colab

**Describe the current behavior**

I have tried to build a Keras model for sentiment analysis using ``tf.GradientTape`` and ``tf.function`` instead of using keras's ``model.compile`` and ``model.fit``. With the latter, the accuracy starts off at around ``0.50`` and raises to ``0.95`` on the training set and ``0.86`` on the test set. While the former just wanders around ``0.50`` accuracy.

**Describe the expected behavior**

It is expected that the accuracy goes up and not just oscillate around ``0.5``.

**Standalone code to reproduce the issue** 

The full notebook: https://colab.research.google.com/drive/1DESzPRktTkYzZ0nnOo9ofNMPaVg-0UHd

The Model:

```python
class ANNForSentimentAnalysis(tf.keras.Model):
    def __init__(self, embedding = "https://tfhub.dev/google/nnlm-en-dim128/1", name="ANNForSentimentAnalysis", **kwargs):
        super(ANNForSentimentAnalysis, self).__init__(name=name, **kwargs)
        self._layers = [
            hub.KerasLayer(embedding, trainable=True, dtype=tf.string),
            Dense(16, activation='relu'),
            Dense(1, activation='sigmoid')
        ]
        # self._model = Sequential(self._layers)

    @tf.function
    def call(self, inputs):
        # return self._model(inputs)
        for layer in self._layers:
            inputs = layer(inputs)
        return inputs

def train_step(train_data, model, optimizer, loss_func, metric):
    for i, batch in enumerate(train_data):
        x_train, y_train = batch
        with tf.GradientTape() as tape:
            preds = model(x_train)
            loss = loss_func(y_train, preds)
        grads = tape.gradient(loss, model.trainable_variables)
        optimizer.apply_gradients(zip(grads, model.trainable_variables))
        metric_eval = metric(y_train, preds)
        sys.stdout.write(f"\rStep {i}: [" + i*"*" + f"]\tloss: {loss:.4f}\taccuracy: {metric_eval:.4f}")
    print("")

def accuracy(y_train, y_pred):
    return tf.reduce_mean(tf.cast(tf.cast(y_train, tf.float32) == tf.cast(y_pred > 0.5, tf.float32), tf.float32))
```

**Training Log**

```none
Step 29: [*****************************]	loss: 0.6937	accuracy: 0.5776
Step 29: [*****************************]	loss: 0.6667	accuracy: 0.4945
Step 29: [*****************************]	loss: 0.6421	accuracy: 0.4564
Step 29: [*****************************]	loss: 0.6559	accuracy: 0.5151
Step 29: [*****************************]	loss: 0.6455	accuracy: 0.5210
Step 29: [*****************************]	loss: 0.6347	accuracy: 0.5201
Step 29: [*****************************]	loss: 0.6236	accuracy: 0.5031
Step 29: [*****************************]	loss: 0.6119	accuracy: 0.5059
Step 29: [*****************************]	loss: 0.5719	accuracy: 0.5007
Step 29: [*****************************]	loss: 0.5685	accuracy: 0.4990
Step 29: [*****************************]	loss: 0.5626	accuracy: 0.4990
Step 29: [*****************************]	loss: 0.5764	accuracy: 0.5074
Step 29: [*****************************]	loss: 0.5735	accuracy: 0.4994
Step 29: [*****************************]	loss: 0.5386	accuracy: 0.4995
Step 29: [*****************************]	loss: 0.5384	accuracy: 0.4995
Step 29: [*****************************]	loss: 0.5517	accuracy: 0.5000
Step 29: [*****************************]	loss: 0.5328	accuracy: 0.4997
Step 29: [*****************************]	loss: 0.5351	accuracy: 0.4979
Step 29: [*****************************]	loss: 0.5574	accuracy: 0.5052
Step 29: [*****************************]	loss: 0.5166	accuracy: 0.5010
```

**Comment by Saduf2019** [comment]
@tirthasheshpatel 
could you please let us know what version of tensorflow are you facing this issue in

**Comment by tirthasheshpatel** [comment]
Tensorflow V2.2

**Comment by ymodak** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.
Thanks!
